pkg/devfile/adapters/docker/utils/utils.go

```diff
@@ -9,18 +9,48 @@ import (
 	"github.com/docker/docker/api/types"
 	"github.com/docker/docker/api/types/container"
 	"github.com/docker/docker/api/types/mount"
+
+	adaptersCommon "github.com/openshift/odo/pkg/devfile/adapters/common"
 	"github.com/openshift/odo/pkg/devfile/parser/data/common"
 	"github.com/openshift/odo/pkg/lclient"
+	"github.com/openshift/odo/pkg/log"
+	"github.com/openshift/odo/pkg/util"
+
+	"github.com/golang/glog"
+	"github.com/pkg/errors"
+)
+
+const (
+	supervisordVolume = "supervisord"
+	projectsVolume    = "projects"
 )
 
 // ComponentExists checks if Docker containers labeled with the specified component name exists
 func ComponentExists(client lclient.Client, name string) bool {
+	containers := GetComponentContainers(client, name)
+	return len(containers) != 0
+}
+
+// GetComponentContainers returns a list of the running component containers
+func GetComponentContainers(client lclient.Client, name string) (containers []types.Container) {
```

Comment:
IMO this component should also return `err` considering that we are using:

```go
containerList, err := client.GetContainerList()
```
Essentially, if this function errors out, we "lose" the error output from client.GetContainerList since all we return is []types.Container of len == 0